okhttp/src/main/java/okhttp3/internal/platform/android/Android10CertificateChainCleaner.kt

```diff
@@ -48,7 +48,8 @@ internal class Android10CertificateChainCleaner(
   }
 
   override fun equals(other: Any?): Boolean =
-      other is Android10CertificateChainCleaner // All instances are equivalent.
+      other is Android10CertificateChainCleaner &&
+          other.trustManager === this.trustManager
 
-  override fun hashCode(): Int = 1
+  override fun hashCode(): Int = System.identityHashCode(trustManager.hashCode())
```

Comment:
Why not `trustManager.hashCode()`

Reply:
To match the use of === above. I can change above to use == and therefore equals. 

They are effectively the same as far as I know. Implementations don’t seem to implement hashcode or equals.

Reply:
Got it. You definitely don't want the identity hash code of the hash code, that's just a random value. I think probably regular equals and hashCode is best, that way you're not baking assumptions of how these classes are implemented.

Reply:
Thanks, typo.